crates/ruff_python_formatter/src/statement/stmt_with.rs

```diff
@@ -98,14 +108,52 @@ impl FormatNodeRule<StmtWith> for FormatStmtWith {
     }
 }
 
+/// Returns `true` if the `with` items should be parenthesized, if at least one item expands.
+///
+/// Black parenthesizes `with` items if there's more than one item and they're already
+/// parenthesized, _or_ there's a single item with a trailing comma.
+fn should_parenthesize(with: &StmtWith, context: &PyFormatContext) -> FormatResult<bool> {
+    if has_magic_trailing_comma(with, context) {
+        return Ok(true);
+    }
+
+    if with.items.len() > 1 && are_with_items_parenthesized(with, context)? {
+        return Ok(true);
+    }
```

Comment:
`are_with_items_parenthesized` seems to return early if there are less than two items. Should we remove the `with.items.len()` check here?